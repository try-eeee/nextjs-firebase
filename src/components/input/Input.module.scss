.input {
  width: 100%;
  padding: 0.75rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  margin-bottom: 0.5rem;
  font-size: 1rem;

  &:disabled {
    background-color: #f5f5f5;
    color: #cacaca;
    cursor: not-allowed;
  }
}
