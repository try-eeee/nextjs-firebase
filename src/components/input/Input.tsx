"use client";

import { UseFormRegisterReturn } from "react-hook-form";
import styles from "./Input.module.scss";

interface InputProps extends React.ComponentPropsWithoutRef<"input"> {
  register?: UseFormRegisterReturn;
}

export function Input(props: InputProps) {
  const { register, ...rest } = props;
  return <input {...register} {...rest} className={styles.input} />;
}
