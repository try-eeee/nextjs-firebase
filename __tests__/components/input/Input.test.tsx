import React from "react";
import { render } from "@testing-library/react";
import { Input } from "@/components/input/Input";

describe("Input UnitTest", () => {

  test("Snapshot Test Of Input", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });
});
