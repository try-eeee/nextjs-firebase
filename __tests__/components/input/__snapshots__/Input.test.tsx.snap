// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input UnitTest Snapshot Test Of Input 1`] = `
<DocumentFragment>
  <input
    class="input"
  />
</DocumentFragment>
`;
